pages/api/avatar/[name].tsx

```diff
@@ -11,7 +11,7 @@ export default async function (req: NextRequest) {
   const name = url.searchParams.get("name");
   const text = url.searchParams.get("text");
   const size = Number(url.searchParams.get("size") || "120");
-  const [username, type] = name?.split(".") || [];
+  const { 0: username, length, [length - 1]: type }: { length: number, [key: number]: string } = name?.split(".") || [];
```

Comment:
```suggestion
  const lastDot = name?.lastIndexOf(".");
  const username = name?.slice(0, lastDot);
  const type = name?.slice(lastDot + 1);
```

Let's try this so that all the other dots are included in the username.